src/InProcessTestHost/Sidecar/AsyncManualResetEvent.cs

```diff
@@ -1,13 +1,24 @@
-﻿// Copyright (c) Microsoft Corporation.
+// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
 namespace Microsoft.DurableTask.Sidecar;
 
+/// <summary>
+/// An asynchronous manual reset event implementation.
+/// </summary>
+/// <remarks>
+/// This class provides an asynchronous version of ManualResetEvent that can be used
+/// for synchronization in async/await scenarios.
+/// </remarks>
 class AsyncManualResetEvent
```

Comment:
Can I have some more context on what this is? The comment just let's me know it's an async version of a thing that I also don't know what does :)